tests/h/conftest.py

```diff
@@ -70,10 +71,23 @@ def cli():
 
 
 @pytest.fixture(scope="session")
-def db_engine():
+def db_engine(tmp_path_factory):
     """Set up the database connection and create tables."""
     engine = sqlalchemy.create_engine(TEST_DATABASE_URL)
-    db.init(engine, should_create=True, should_drop=True, authority=TEST_AUTHORITY)
+
+    shared_tmpdir = tmp_path_factory.getbasetemp().parent
+    done_file = shared_tmpdir / "db_initialized.done"
+    lock_file = shared_tmpdir / "db_initialized.lock"
+
+    with FileLock(str(lock_file)):
+        if done_file.is_file():
+            pass
+        else:
+            db.init(
+                engine, should_create=True, should_drop=True, authority=TEST_AUTHORITY
+            )
+            done_file.touch()
```

Comment:
Make only one process initialize the DB and make the other processes wait until the DB is initialized before proceeding, same as done in LMS. I should add some comments to this as in LMS...